checklist/src/main/scala/checklist/Rule.scala

```diff
@@ -4,20 +4,36 @@ import cats.{Applicative, Traverse}
 import cats.data.Ior
 import cats.instances.all._
 import cats.syntax.all._
-import monocle.{PLens, Lens}
+import monocle.PLens
 import scala.language.higherKinds
 import scala.util.matching.Regex
-import Message.{errors, warnings}
+import Message.errors
+import cats.data.NonEmptyList
 
 sealed abstract class Rule[A, B] {
   def apply(value: A): Checked[B]
 
   def map[C](func: B => C): Rule[A, C] =
     Rule.pure(value => this(value) map func)
 
+  def emap[C](func: B => Checked[C]): Rule[A, C] =
```

Comment:
I've not seen this name elsewhere. Is it a name with provenance, or is there a clearer alternative?

Reply:
I stole it from [circe](https://github.com/circe/circe/blob/master/modules/core/shared/src/main/scala/io/circe/Decoder.scala#L193). I'm not sure where they got the name though.  I'm happy to call it whatever.. I'm really not sure what a good and proper name would be for this is.

Reply:
Ok. Let's merge it now and rename it on a point release if it causes confusion.